lib/matplotlib/transforms.py

```diff
@@ -896,8 +949,9 @@ def update_from_data_x(self, x, ignore=None):
            - When ``None``, use the last value passed to :meth:`ignore`.
         """
         x = np.ravel(x)
-        self.update_from_data_xy(np.column_stack([x, np.ones(x.size)]),
-                                 ignore=ignore, updatey=False)
+        xy = np.empty((x.size, 2), dtype=x.dtype)
+        xy[:, 0] = x
+        self.update_from_data_xy(xy, ignore=ignore, updatey=False)
```

Comment:
I think duplicating the array is even a little faster (but mostly simpler) with a quick timeit. 114us compared to 120us so not a major difference.
`python -m timeit "import numpy as np; x=np.linspace(0, 100, 100000); y=np.array([x, x]).T"`
`python -m timeit "import numpy as np; x=np.linspace(0, 100, 100000); y=np.empty((x.size, 2)); y[:, 0]=x"`
Not sure if there is a way to duplicate the view into the new dimension.
```suggestion
        self.update_from_data_xy(np.array([x, x]).T, ignore=ignore, updatey=False)
```

Reply:
I do like this better, and since the change needed to be made to check for NaNs in both columns, `np.empty` was no longer appropriate since it could have inf or NaNs.